src/client/dfs/dfs.c

```diff
@@ -1153,39 +1147,20 @@ dfs_mount(daos_handle_t poh, daos_handle_t coh, int flags, dfs_t **_dfs)
 	if (rc != 0)
 		D_GOTO(err_dfs, rc = daos_der2errno(rc));
 
-	prop = daos_prop_alloc(0);
-	if (prop == NULL) {
-		D_ERROR("Failed to allocate prop.");
-		D_GOTO(err_dfs, rc = ENOMEM);
-	}
-
-	rc = daos_pool_query(poh, NULL, &pool_info, prop, NULL);
-	if (rc) {
-		D_ERROR("daos_pool_query() Failed (%d)\n", rc);
-		daos_prop_free(prop);
-		D_GOTO(err_dfs, rc = daos_der2errno(rc));
-	}
-
 	/* Convert the owner information to uid/gid */
-	entry = daos_prop_entry_get(prop, DAOS_PROP_PO_OWNER);
+	entry = daos_prop_entry_get(prop, DAOS_PROP_CO_OWNER);
 	D_ASSERT(entry != NULL);
 	rc = daos_acl_principal_to_uid(entry->dpe_str, &dfs->uid);
-	if (rc != 0) {
-		D_ERROR("Unable to convert owner to uid\n");
-		daos_prop_free(prop);
-		D_GOTO(err_dfs, rc = daos_der2errno(rc));
-	}
+	if (rc != 0)
+		/** Set uid to nobody */
+		dfs->uid = 65534;
```

Comment:
65534 is a common uid/gid for nobody, but [it looks like the implementation can vary by distro](https://stackoverflow.com/questions/34831861/can-i-assume-that-nobody-is-65534). You could run daos_acl_principal_to_uid/gid on the name "nobody@" instead to make sure it always gets the right value.

Reply:
what if it is nfsnobody? is nobody always there?

Reply:
[According to the Linux standard spec](http://refspecs.linux-foundation.org/LSB_3.1.0/LSB-Core-generic/LSB-Core-generic/usernames.html), "nobody" is an optional user, but it is at least a standard name defined there. In theory it might not exist--at that point maybe we could bomb out, because we have no one to assign the files to.